crates/solver/src/driver.rs

```diff
@@ -435,13 +452,33 @@ impl Driver {
                     .map_err(|err| anyhow!("{err}"))
                     .context("convert nonce")?,
             };
+            let scores = model::solver_competition::Scores {
+                winning_score: scores.last().copied().expect("no score"), // guaranteed to exist
+                // reference score is the second highest score, or 0 if there is only one score (see
+                // CIP20)
+                reference_score: scores.get(scores.len() - 2).copied().unwrap_or(0.into()),
```

Comment:
You could not `.collect()` the `scores` iterator and instead `.pop()` it twice to get the winner and runner-up. Just bringing it up to avoid indexing which can always a bit error prone.

Reply:
True, ended up with nicer code.